packages/dds/tree/src/typeCheckTests.ts

```diff
@@ -0,0 +1,210 @@
+	| requireTrue<isAssignableTo<GenericBivariant<Nominal1>, GenericBivariant<Derived1>>>
+
+	// test Invariant
+	| requireFalse<isAssignableTo<GenericInvariant<Nominal1>, GenericInvariant<Nominal2>>>
+	| requireFalse<isAssignableTo<GenericInvariant<Derived1>, GenericInvariant<Nominal1>>>
+	| requireFalse<isAssignableTo<GenericInvariant<Nominal1>, GenericInvariant<Derived1>>>
+
+	// test Multiple parameters
+	| requireFalse<isAssignableTo<GenericMulti<Nominal1, number>, GenericMulti<Derived1, number>>>
+	| requireFalse<isAssignableTo<GenericMulti<number, Nominal1>, GenericMulti<number, Derived1>>>
+	| requireTrue<isAssignableTo<GenericMulti<number, Derived1>, GenericMulti<number, Nominal1>>>
+
+	// test Covariant Interface
+	| requireFalse<isAssignableTo<GenericCovariantInterface<Nominal1>, GenericCovariantInterface<Nominal2>>>
+	| requireTrue<isAssignableTo<GenericCovariantInterface<Derived1>, GenericCovariantInterface<Nominal1>>>
+	| requireFalse<isAssignableTo<GenericCovariantInterface<Nominal1>, GenericCovariantInterface<Derived1>>>
+
+	// test Contravariant Interface
+	| requireFalse<isAssignableTo<GenericContravariantInterface<Nominal1>, GenericContravariantInterface<Nominal2>>>
+	| requireFalse<isAssignableTo<GenericContravariantInterface<Derived1>, GenericContravariantInterface<Nominal1>>>
+	| requireTrue<isAssignableTo<GenericContravariantInterface<Nominal1>, GenericContravariantInterface<Derived1>>>
+
+	// test Bivariant Interface
+	| requireFalse<isAssignableTo<GenericBivariantInterface<Nominal1>, GenericBivariantInterface<Nominal2>>>
+	| requireTrue<isAssignableTo<GenericBivariantInterface<Derived1>, GenericBivariantInterface<Nominal1>>>
+	| requireTrue<isAssignableTo<GenericBivariantInterface<Nominal1>, GenericBivariantInterface<Derived1>>>
+
+	// test Invariant Interface
+	| requireFalse<isAssignableTo<GenericInvariantInterface<Nominal1>, GenericInvariantInterface<Nominal2>>>
+	| requireFalse<isAssignableTo<GenericInvariantInterface<Derived1>, GenericInvariantInterface<Nominal1>>>
+	| requireFalse<isAssignableTo<GenericInvariantInterface<Nominal1>, GenericInvariantInterface<Derived1>>>
+
+	// test eitherIsAny
+	| requireTrue<eitherIsAny<any, Nominal1>>
+	| requireTrue<eitherIsAny<Nominal1, any>>
+	| requireTrue<eitherIsAny<any, any>>
+	| requireFalse<eitherIsAny<Nominal1, Nominal1>>
+
+	// areSafelyAssignable tests
+	| requireTrue<areSafelyAssignable<Nominal1, Nominal1>>
+	| requireFalse<areSafelyAssignable<unknown, Nominal1>>
+	| requireFalse<areSafelyAssignable<any, any>>
+	| requireTrue<areSafelyAssignable<unknown, unknown>>
+	| requireFalse<areSafelyAssignable<any, Nominal1>>
+	| requireFalse<areSafelyAssignable<Nominal1, any>>
+	| requireFalse<areSafelyAssignable<unknown, any>>
+
+	// test isAny
+	| requireTrue<isAny<any>>
+	| requireFalse<isAny<unknown>>
+	| requireFalse<isAny<Nominal1>>
+	| requireFalse<isAny<never>>
+	| requireFalse<isAny<{}>>
+
```

Comment:
`requireFalse<isAny<boolean>>`, for good measure? Then we are testing a non-object primitive and also the type used in the implementation of `isAny` (not that it should matter, but appears like it might superficially).